asteroidsyncserviced/bluez/bluezclient.cpp

```diff
@@ -18,29 +18,38 @@
  */
 
 #include "bluezclient.h"
+#include "bluezclient_p.h"
 #include "dbus-shared.h"
-#include "../libasteroid/services/common.h"
+#include "services/common.h"
+
+#include "bluez_helper.h"
 
 #include <QDBusConnection>
 #include <QDBusReply>
 #include <QDebug>
+#include <stdexcept>
 
-BluezClient::BluezClient(QObject *parent):
-    QObject(parent),
+BluezClientPrivate::BluezClientPrivate():
     m_dbus(QDBusConnection::systemBus()),
     m_bluezManager("org.bluez", "/", m_dbus)
+{
+}
+
+BluezClient::BluezClient(QObject *parent):
+    QObject(parent),
+    d_ptr{std::make_unique<BluezClientPrivate>()}
 {
     qDBusRegisterMetaType<InterfaceList>();
     qDBusRegisterMetaType<ManagedObjectList>();
 
-    if (m_bluezManager.isValid()) {
-        connect(&m_bluezManager, SIGNAL(InterfacesAdded(QDBusObjectPath,InterfaceList)),
+    if (d_ptr && d_ptr->m_bluezManager.isValid()) {
+        connect(&d_ptr->m_bluezManager, SIGNAL(InterfacesAdded(QDBusObjectPath,InterfaceList)),
                 this, SLOT(slotInterfacesAdded(QDBusObjectPath,InterfaceList)));
 
-        connect(&m_bluezManager, SIGNAL(InterfacesRemoved(QDBusObjectPath,QStringList)),
+        connect(&d_ptr->m_bluezManager, SIGNAL(InterfacesRemoved(QDBusObjectPath,QStringList)),
                 this, SLOT(slotInterfacesRemoved(QDBusObjectPath,QStringList)));
 
-        auto objectList = m_bluezManager.GetManagedObjects().argumentAt<0>();
+        auto objectList = d_ptr->m_bluezManager.GetManagedObjects().argumentAt<0>();
```

Comment:
Could you keep things consistent by using `d_func` here too ? (just like in `BluezClient::pairedWatches`)

Reply:
Yes, all raw `d_ptr` uses that were convertible to use `d_func()` now do so.